File: backend/src/main/java/tko/model/dto/workout/WorkoutForPageDTO.java
package tko.model.dto.workout;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutForPageDTO {
    private Long id;
    private String name;
    private Integer likeCount;
    private boolean liked;
}

File: android/app/src/main/java/tk/ssau/fitnesstko/model/dto/WorkoutForPageDto.kt
package tk.ssau.fitnesstko.model.dto

/**
 * Упрощённое представление тренировки для списков/превью.
 * @param id Уникальный идентификатор тренировки.
 * @param name Название тренировки.
 * @param likeCount Общее количество лайков.
 * @param liked Лайкнул ли текущий пользователь эту тренировку.
 */
data class WorkoutForPageDto(
    val id: Long?,
    val name: String?,
    val likeCount: Int?,
    val liked: Boolean
)
